From ede47ad227e82bf90b089f58d55230ceb5db2667 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Wed, 24 Jul 2013 18:22:55 +0300
Subject: [PATCH 301/987] dpaa_eth: remove unused bpool depletion callback

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

The stub bpool depletion callback is not used, removed.

Change-Id: Ibf465d4c04bc867a7cda8a92c7cf07b3ae682de0

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: Ie864646adbe77817b59fc9cf8223677a4048be2a
Reviewed-on: http://git.am.freescale.net:8181/3600
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Sovaiala Cristian-Constantin-B39531 <Cristian.Sovaiala@freescale.com>
Reviewed-by: Radulescu Ruxandra Ioana-B05472 <ruxandra.radulescu@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.c        | 11 ++---------
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c | 11 +----------
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h |  2 --
 3 files changed, 3 insertions(+), 21 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
index 3fce5ee..6ac088c 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
@@ -669,19 +669,13 @@ dpa_priv_bp_alloc(struct dpa_bp *dpa_bp)
 	BUG_ON(dpa_bp->size == 0);
 	BUG_ON(dpa_bp->config_count == 0);
 
-	bp_params.flags = BMAN_POOL_FLAG_DEPLETION;
-	bp_params.cb = dpa_bp_depletion;
-	bp_params.cb_ctx = dpa_bp;
+	bp_params.flags = 0;
 
 	if (default_pool) {
 		atomic_inc(&default_pool->refs);
 		return 0;
 	}
 
-	/* If the pool is already specified, we only create one per bpid */
-	if (dpa_bpid2pool_use(dpa_bp->bpid))
-		return 0;
-
 	if (dpa_bp->bpid == 0)
 		bp_params.flags |= BMAN_POOL_FLAG_DYNAMIC_BPID;
 	else
@@ -708,8 +702,7 @@ dpa_priv_bp_alloc(struct dpa_bp *dpa_bp)
 
 	dpa_bp->dev = &pdev->dev;
 
-	if (!default_pool)
-		default_pool = dpa_bp;
+	default_pool = dpa_bp;
 
 	dpa_bpid2pool_map(dpa_bp->bpid, dpa_bp);
 
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
index 4421765..e1b324e 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.c
@@ -753,13 +753,6 @@ _return_of_node_put:
 	return dpa_bp;
 }
 
-void dpa_bp_depletion(struct bman_portal	*portal,
-		struct bman_pool *pool, void *cb_ctx, int depleted)
-{
-	if (net_ratelimit())
-		pr_err("Invalid Pool depleted notification!\n");
-}
-
 static void dpaa_eth_seed_pool(struct dpa_bp *bp)
 {
 	int count = bp->target_count;
@@ -817,9 +810,7 @@ dpa_bp_alloc(struct dpa_bp *dpa_bp)
 	BUG_ON(dpa_bp->size == 0);
 	BUG_ON(dpa_bp->config_count == 0);
 
-	bp_params.flags = BMAN_POOL_FLAG_DEPLETION;
-	bp_params.cb = dpa_bp_depletion;
-	bp_params.cb_ctx = dpa_bp;
+	bp_params.flags = 0;
 
 	/* If the pool is already specified, we only create one per bpid */
 	if (dpa_bpid2pool_use(dpa_bp->bpid))
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h
index 6c47496..9c9fe3b 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h
@@ -90,8 +90,6 @@ int dpa_bp_create(struct net_device *net_dev, struct dpa_bp *dpa_bp,
 		size_t count);
 void __cold __attribute__((nonnull))
 dpa_bp_free(struct dpa_priv_s *priv, struct dpa_bp *dpa_bp);
-void dpa_bp_depletion(struct bman_portal	*portal,
-		struct bman_pool *pool, void *cb_ctx, int depleted);
 struct dpa_bp *dpa_bpid2pool(int bpid);
 void dpa_bpid2pool_map(int bpid, struct dpa_bp *dpa_bp);
 bool dpa_bpid2pool_use(int bpid);
-- 
1.9.1

